fix(DatePicker): explicitly update minDate and maxDate

figfofu opened the pull request:
Since flatpickr does not automatically update minDate and maxDate,
we need to explicitly do it when the user makes any change in the props
by using set(option, value) method from flatpickr

Fixes #2500 

#### Changelog

**New**

**Changed**

- explicitly update minDate and maxDate when they are changed

**Removed**

#### Testing / Reviewing

Testing should make sure <DatePicker> is not broken.

Commit:
fix(DatePicker): explicitly update minDate and maxDate

Since flatpickr does not automatically update minDate and maxDate,
we need to explicitly do it when the user makes any change in the props
by using set(option, value) method from flatpickr

asudoh reviewed (commented):
Hi 👋 @figfofu thank you for jumping in! Any change to align this change to #4856? Thanks!

figfofu commented:
I think we can do that.
I was thinking of making the change to either `componentDidUpdate()` or `componentWillUpdate()`,
and then decided to add to the existing method `componentWillUpdate()`.

I can wait for #4856 getting merged then add the change.

Commit:
fix(DatePicker): explicitly update minDate and maxDate

Since flatpickr does not automatically update minDate and maxDate,
we need to explicitly do it when the user makes any change in the props
by using set(option, value) method from flatpickr
Aligned to the change in #4856

asudoh commented:
Thank you for the update @figfofu! Going through props with an array is an interesting approach, but I'd rather not abstracting that... Would it be OK for you to add the logic for `minDate`/`maxDate` to the existing `cDU`?

BTW you can simply add comments once you add update commits and I'll get notified via email.

Commit:
fix(DatePicker): explicitly update minDate and maxDate

Since flatpickr does not automatically update minDate and maxDate,
we need to explicitly do it when the user makes any change in the props
by using set(option, value) method from flatpickr
Aligned to the change in #4856

figfofu commented:
@asudoh 
Updated the change.
Please review it.

asudoh reviewed (approved):
LGTM :+1: - Thanks @figfofu!

joshblack committed:
Merge branch 'master' into date_picker_update_minDate_maxDate
